To: vim_dev@googlegroups.com
Subject: Patch 7.3.458
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.458
Problem:    Crash when calling smsg() during startup.
Solution:   Don't use 'shortmess' when it is not set yet.
Files:	    src/option.c


*** ../vim-7.3.457/src/option.c	2012-02-20 22:18:22.000000000 +0100
--- src/option.c	2012-02-26 13:14:48.000000000 +0100
***************
*** 10984,10990 ****
  shortmess(x)
      int	    x;
  {
!     return (   vim_strchr(p_shm, x) != NULL
  	    || (vim_strchr(p_shm, 'a') != NULL
  		&& vim_strchr((char_u *)SHM_A, x) != NULL));
  }
--- 10984,10991 ----
  shortmess(x)
      int	    x;
  {
!     return p_shm != NULL &&
! 	    (   vim_strchr(p_shm, x) != NULL
  	    || (vim_strchr(p_shm, 'a') != NULL
  		&& vim_strchr((char_u *)SHM_A, x) != NULL));
  }
*** ../vim-7.3.457/src/version.c	2012-02-29 13:49:03.000000000 +0100
--- src/version.c	2012-02-29 13:50:52.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     458,
  /**/

-- 
Support your right to bare arms!  Wear short sleeves!

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
